### PR TITLE
Feature stars

### PR DESCRIPTION
**Pull request by marc-jones**
__asteroid_out_of_bounds and __star_out_of_bounds can probably be combined as additional arguments can be passed to the onOutOfBounds event listener (http://phaser.io/docs/2.4.6/Phaser.Signal.html#add)

EDIT: Done!

### CODE REVIEW

File: src/states/game.js [hunk 1]
```diff
@@ -334,23 +396,21 @@ Game.prototype =
         return this.__uuid;
     },
 
-    __asteroid_out_of_bounds : function(asteroid)
+    __object_out_of_bounds : function(object, object_array_string)
     {
-        if (asteroid.position.x > CONSTS.SCREEN_DIMENSIONS[1])
+        if (object.position.x > CONSTS.SCREEN_DIMENSIONS[0])
```
Comment (marc-jones) [inline src/states/game.js:401]: I think this was a bug?

**Comment by marc-jones** [comment]
Don't know how the initial two commits got replicated...

File: src/states/game.js [hunk 2]
```diff
@@ -115,12 +127,28 @@ Game.prototype =
         {
             this.debug_graphics = this.game.add.graphics();
         }
+        
+        for (size = 1; size < 4; size++) {
```
Comment (tomcraven) [inline src/states/game.js]: Here, size is created in global scope. if you type window.size into the console you can see it. Not a massive problem but we can easily avoid this with `for (var size = 1...`

File: src/states/game.js (hunk 1 above)
Comment (tomcraven) [inline src/states/game.js:401]: woops, yes i think it was, good eye

File: src/states/game.js [hunk 3]
```diff
@@ -334,23 +396,21 @@ Game.prototype =
...
+        if (object.position.x > CONSTS.SCREEN_DIMENSIONS[0])
         {
             return;
         }
-
-        asteroid.destroy();
-        this.__non_player_objects =
-            this.__non_player_objects.filter(function(item)
+        object.destroy();
+        this[object_array_string] =
```
Comment (tomcraven) [inline src/states/game.js:406]: This feels a bit naughty haha, can we not pass the object array in?

File: src/states/game.js (hunk 2 above)
Comment (marc-jones) [inline src/states/game.js]: Ah, thanks, it's these subtleties of Javascript that I don't know.

File: src/states/game.js (hunk 3 above)
Comment (marc-jones) [inline src/states/game.js:406]: I tried passing this.__stars and this.__asteroids but I couldn't get them to update globally. Maybe you can show me how it's done?!

Comment (tomcraven) [inline src/states/game.js:406]: Ah hmm I see what you mean, I'll give it a go...

Comment (marc-jones) [inline src/states/game.js:406]: Any luck?

Comment (tomcraven) [inline src/states/game.js:406]: Hmm no, not at all! Merge away :)

**Pull request merged**